frontend/lib/qr_code_scanner/qr_code_scanner.dart

```diff
@@ -180,41 +180,34 @@ class _QRViewState extends State<QRCodeScanner> {
     });
   }
 
-  void _tryParseCodeContent(String codeContent) {
-    try {
-      final cardDetails = parseQRCodeContent(codeContent);
-      // TODO this method gets called multiple times in a row for some reason
-      // The following code should be exclusive, but dart does not natively
-      // support mutexes, synchronized, etc.
-      if (isDone) {
-        return;
-      }
-      isDone = true;
-      Provider.of<CardDetailsModel>(context, listen: false)
-          .setCardDetails(cardDetails);
-      Navigator.of(context).maybePop();
-    } on Exception catch (e) {
+  void _tryParseCodeContent(String codeContent) async {
+    var wasSuccessful = false;
+    // needed because this method gets called multiple times in a row after one
+    // qr code gets detected, therefore we need to protect it
+    if (isDone) {
+      return;
+    }
+    isDone = true;
+    await qrCodeParsingMutex.protect(() async {
       controller.pauseCamera();
-      print("Failed to parse qr code content!");
-      print(e);
-      String errorMessage;
-      if (e is InvalidJwtException) {
-        errorMessage =
-            "Der Inhalt des QR-Codes kann von der App nicht verstanden werden.";
+      final parseResult = widget.qrCodeContentParser(codeContent);
+      if (parseResult.hasError) {
+        print("Failed to parse qr code content!");
+        print(parseResult.internalErrorMessage);
+        final errorMessage = parseResult.userErrorMessage;
+        _showErrorDialog(errorMessage).then((value) {
+          Future.delayed(Duration(milliseconds: scanDelayAfterErrorMs))
```

Comment:
Why should we delay this? Maybe add a comment

Reply:
To give the user time to move the phone away from the qr code. Otherwise we directly scan the broken qr code again if the phone wasn't moved

Reply:
Ah okay, could you add a comment?